safe_eth/safe/safe_signature.py

```diff
@@ -488,33 +536,80 @@ def is_valid(
 
 
 class SafeSignatureContractAsync(SafeSignatureContractMixin, SafeSignatureAsync):
+    async def _check_eip1271(
+        self,
+        web3: AsyncWeb3,
+        handler_getter,
+        function_signature: str,
+        data: bytes,
+        signature: bytes,
+    ) -> bool:
+        """
+        Attempt to validate an EIP-1271 signature using a specific CompatibilityFallbackHandler and function signature.
+
+        :param web3: AsyncWeb3 instance
+        :param handler_getter: A function that returns the appropriate CompatibilityFallbackHandler contract for the current Safe.
+        :param function_signature: The ABI function signature to call
+        :param data: The data or hash to be validated, depending on the Safe version.
+        :param signature: The contract signature payload to validate.
+        :return: True on successful validation; otherwise False.
+        """
+        handler = handler_getter(web3, self.owner)
```

Comment:
Not liking duplication of cod, but also I cannot think of an easiest way to do it